lib/create-app.js

```diff
@@ -187,30 +204,20 @@ function initWebId (argv, app, ldp) {
   // Set up authentication-related API endpoints and app.locals
   initAuthentication(app, argv)
 
-  // Protect against requests from third-party applications
-  app.use((req, res, next) => {
-    // Reject cookies from third-party applications.
-    // Otherwise, when a user is logged in to their Solid server,
-    // any third-party application could perform authenticated requests
-    // without permission by including the credentials set by the Solid server.
-    const origin = req.headers.origin
-    const userId = req.session.userId
-    if (!argv.host.allowsSessionFor(userId, origin)) {
-      debug(`Rejecting session for ${userId} from ${origin}`)
-      // Destroy session data
-      delete req.session.userId
-      // Ensure this modified session is not saved
-      req.session.save = done => done()
-    }
-    next()
-  })
-
-  // Set up per-host LDP middleware
   if (argv.multiuser) {
     app.use(vhost('*', LdpMiddleware(corsSettings)))
   }
 }
 
+/**
+ * Determines whether the given request is a logout request
+ */
+function isLogoutRequest (req) {
+  // TODO: this is a hack that hard-codes OIDC paths,
+  // this code should live in the OIDC module
+  return req.path === '/logout' || req.path === '/goodbye'
```

Comment:
@kidehen I copied this code from @smalinin, but I actually don't think that goodbye is necessary in this exception list (although it doesn't really hurt). The user needs to be authenticated for logout, but not for goodbye, right?

Reply:
Hmm..
I would suggest the workaround as is for now, we know it works (and it doesn't hurt  in the worst case).  

/cc @smalinin @cblakeley